src/Microsoft.ML.FastTree/Dataset/IntArray.cs

```diff
@@ -32,6 +34,25 @@ internal abstract class IntArray : IEnumerable<int>
         /// </summary>
         public abstract int Length { get; }
 
+        // We can test if the dll is present by calling into a non-existant method.
+        // If we get an EntryPointNotFoundException then the dll exists and we can use the native code.
+        // If we get a DllNotFoundException then the dll doesn't exist and we should use the managed fallbacks.
+        private const string NativePath = "FastTreeNative";
+        [DllImport(NativePath), SuppressUnmanagedCodeSecurity]
+        private static extern void NonExistantMethod();
+
+        /// <summary>
+        /// Lazy bool that checks if we are on x86/x64 so we know if we should use the native code
+        /// or the managed fallbacks.
+        /// </summary>
+        public static Lazy<bool> UseFastTreeNative = new(() =>
```

Comment:
This doesn't need to be lazy. It can just be

```c#
public static bool UseFastTreeNative => RuntimeInformation.ProcessArchitecture == Architecture.X64 || RuntimeInformation.ProcessArchitecture == Architecture.X86
```

Reply:
Sounds good. Has been switched.